test/integration/targets/test_uri/tasks/main.yml

```diff
@@ -0,0 +1,26 @@
+- name: Assert url tests
+  assert:
+    that:
+      - "'http://google.com' is url"      # simple case
+      - "'htp://google.com' is not url"   # bad schema
+      - "'ftp://google.com' is url"       # other schema
+      - "'file://etc/hosts' is not url"   # uris not urls
+      - "'mailto://me@nowhere.com' is not url"
+      - "'asldkfjhalsidfthjo' is not url" # junk
+      - "'http://admin:secret@example.com' is url"
+      - "'ftps://admin:secret@example.com' is url"
+      - "'admin:secret@example.com' is not url"
+      - "'http://admin:secret@example.com/myfile?parm=1&param=2' is url"
+- name: Assert uri tests
+  assert:
+    that:
+      - "'http://google.com' is uri"      # simple case, urls are uris but not all uris are urls
+      - "'ftp://google.com' is uri"       # other schema
+      - "'file://etc/hosts' is uri "      # uris that are not urls
+      - "'mailto://me@nowhere.com' is uri"
+      - "'sftp://me@nowhere.com' is uri"
+      - "'asldkfjhalsidfthjo' is not uri" # junk
+      - "'http://admin:secret@example.com' is uri"
+      - "'ftps://admin:secret@example.com' is uri"
+      - "'admin:secret@example.com' is not uri"
```

Comment:
This is a URI, but not a URL.

Reply:
actually, depends on how you interpret `admin:` as it can be a schema/protocol, in this case we take it as part of auth and missing a schema (though you could argue that http is the implied schema, i was trying to avoid using implied)

If strictly using implied schemas  then `''` is a uri (implied http:// and empty path). But i'm open to suggestions on how to solve this quandary.

Reply:
https://url.spec.whatwg.org/#urls
in this table `example` is both a valid and invalid URL, depending on implied origin in this case. One reason i was trying to avoid implied parts as in the context of the test .. there is no other context (unless we allow passing that context in)